js/utils.js

```diff
@@ -415,6 +416,191 @@ Fliplet.Registry.set('dynamicListUtils', (function() {
     }
   }
 
+  function getInputDate(date, getDate, timeOnly, dateOnly) {
+    var inputDate = null;
+
+    if (!dateOnly) {
+      inputDate = timeOnly
+        ? getDate(date, 'HH:mm')
+        : getDate(date);
+    }
+
+    return inputDate;
+  }
+
+  function getDateModifiedValues(options) {
+    var timestamp = options.date;
+    var timeOnly = /^([0-1][0-9]|[2][0-3]):([0-5][0-9])$/.test(timestamp);
+    var dateOnly = !/[:]/.test(timestamp);
+    var deviceTimeZone = moment.tz.guess();
+    var getDate = options.useDeviceTimezone ? moment.tz.setDefault(deviceTimeZone) : moment.utc;
+
+    switch (options.dateFilterModifiers) {
+      case 'today':
+        return {
+          today: moment().startOf('day'),
```

Comment:
1. `moment().startOf('day')` should also go through the same caching system that the other date modifiers are using. See https://jsbench.me/q5klic37ss for how poor `moment().startOf('day')` performs.
2. `today` isn't really "today". Instead of the values `today` and `inputDate`, let's use `comparisonDate` and `entryDate` instead. `comparisonDate` is the control value for comparison and `entryDate` represents the value from each entry.
3. `getCurrentDate()` should be `getCachedDate()`. `getCachedDate()` will load the value from cache if it exists, or create the value from scratch and save it into cache before returning the value.